lucille-core/src/test/java/com/kmwllc/lucille/stage/ElasticsearchLookupTest.java

```diff
@@ -0,0 +1,212 @@
+package com.kmwllc.lucille.stage;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import java.io.IOException;
+import java.util.List;
+import java.util.function.Function;
+import org.apache.zookeeper.KeeperException.UnimplementedException;
+import org.eclipse.jetty.util.component.Graceful.ThrowingRunnable;
+import org.junit.Test;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.kmwllc.lucille.core.Document;
+import com.kmwllc.lucille.core.StageException;
+import com.kmwllc.lucille.util.ElasticsearchUtils;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import co.elastic.clients.elasticsearch.ElasticsearchClient;
+import co.elastic.clients.elasticsearch.core.GetRequest;
+import co.elastic.clients.elasticsearch.core.GetResponse;
+import co.elastic.clients.transport.ElasticsearchTransport;
+import co.elastic.clients.transport.endpoints.BooleanResponse;
+import co.elastic.clients.util.ObjectBuilder;
+
+public class ElasticsearchLookupTest {
+
+  private void assertMessage(String message, ThrowingRunnable run) throws Exception {
+    try {
+      run.run();
+      fail("Exception not thrown");
+    } catch (StageException e) {
+      assertEquals(message, e.getMessage());
+    }
+
+  }
+
+  @Test
+  public void testMalformedConfigs() throws Exception {
+    Config one = ConfigFactory.load("ElasticSearchLookupTest/noSource.conf");
+    Config two = ConfigFactory.load("ElasticSearchLookupTest/noDest.conf");
+    Config three = ConfigFactory.load("ElasticSearchLookupTest/noUrl.conf");
+    Config four = ConfigFactory.load("ElasticSearchLookupTest/noIndex.conf");
+    assertThrows(IllegalArgumentException.class, () -> new ElasticsearchLookup(one));
+    assertThrows(IllegalArgumentException.class, () -> new ElasticsearchLookup(two));
+    assertThrows(IllegalArgumentException.class, () -> new ElasticsearchLookup(three));
+    assertThrows(IllegalArgumentException.class, () -> new ElasticsearchLookup(four));
+  }
+
+  @Test
+  public void testStart() throws Exception {
+    Config config = ConfigFactory.parseResourcesAnySyntax("ElasticsearchLookupTest/append.conf");
```

Comment:
could you give the configs descriptive names here like you did in testStop(), testProcessDocument(), etc.?